src/contrast.rs

```diff
@@ -83,6 +175,51 @@ mod test {
         assert_eq!(hist[3], 5);
         assert!(hist.iter().skip(4).all(|x| *x == 5));
     }
+    #[test]
+    fn test_histogram(){
+        let image: GrayImage = ImageBuffer::from_raw(5, 1, vec![
+            1u8, 2u8, 3u8, 2u8, 1u8]).unwrap();
+
+        let hist = histogram(&image);
+
+        assert_eq!(hist[0], 0);
+        assert_eq!(hist[1], 2);
+        assert_eq!(hist[2], 2);
+        assert_eq!(hist[3], 1);
+    }
+    #[test]
+    fn test_otsu_level(){
+        let image: GrayImage = ImageBuffer::from_raw(26, 1, 
+            vec![0u8, 10u8, 20u8, 30u8, 40u8, 50u8, 60u8, 70u8, 
+                80u8, 90u8, 100u8, 110u8, 120u8, 130u8, 140u8, 
+                150u8, 160u8, 170u8, 180u8, 190u8, 200u8,  210u8,  
+                220u8,  230u8,  240u8,  250u8]).unwrap();
+        let level = otsu_level(&image);
+        assert_eq!(level, 125);
+    }
+    #[test]
+    fn test_threshold(){
+        let before: GrayImage = ImageBuffer::from_raw(26, 1, 
+            vec![0u8, 10u8, 20u8, 30u8, 40u8, 50u8, 60u8, 70u8, 
+                80u8, 90u8, 100u8, 110u8, 120u8, 130u8, 140u8, 
+                150u8, 160u8, 170u8, 180u8, 190u8, 200u8,  210u8,  
+                220u8,  230u8,  240u8,  250u8]).unwrap();
+        let after: GrayImage = ImageBuffer::from_raw(26, 1, 
+            vec![0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 
+                0u8, 0u8, 0u8, 0u8, 0u8, 255u8, 255u8, 
+                255u8, 255u8, 255u8, 255u8, 255u8, 255u8,  255u8,  
+                255u8,  255u8,  255u8,  255u8]).unwrap();
+
+        let test = threshold(&before, 125u8);
+
+        for y in 0..after.height() {
+            for x in 0..after.width() {
+                let check = after.get_pixel(x, y)[0] as usize;
+                let test_pixel = test.get_pixel(x, y)[0] as usize;
+                assert_eq!(test_pixel, check);
+            }
```

Comment:
assert_pixels_eq will do this more succinctly: assert_pixels_eq!(test, after). I'd use "actual" for "test" and "expected" for "after" here, as it's slightly clearer what each value is for.